### ISSUE TITLE
clj-kondo gives warning about invalid portfolio config

### ISSUE THREAD

**Issue by svdo** [cjohansen/portfolio#22]
Thanks for Portfolio Christian, it's a great project!

When using clj-kondo and portfolio in my project, clj-kondo ends up creating the file `.clj-kondo/no.cjohansen/portfolio/config.edn`, as expected. However, that file is empty, which causes clj-kondo to issue a warning when running the linters:

```
WARNING: error while reading /Users/stefanmm/Documents/Projects/emma-mono/.clj-kondo/no.cjohansen/portfolio/config.edn (EOF while reading)
```

I don't know if the file is intentionally empty (in which case `{}` as contents of the file solves the issue), or if there should be something else in there. It would be awesome if this can be fixed. Thanks!!

**Comment by borkdude** [comment]
I can't repro this with:

```
$ mkdir -p .clj-kondo
$ clj-kondo --lint "$(clojure -Spath -Sdeps '{:deps {no.cjohansen/portfolio {:mvn/version \"2023.12.07\"}}}')" --copy-configs --dependencies
No configs copied.
```

Which deps.edn are you using?

**Comment by svdo** [comment]
Hmm not sure what is going on. The file is actually created when I launch vscode in my repo. In other words: vscode is not running, and in the terminal I type `code .` to launch it. That's enough to trigger the file being created. So I guess it's either Calva or clojure-lsp doing it?

**Comment by borkdude** [comment]
Yes, but it depends on what deps you are using. So,

> Which deps.edn are you using?

**Comment by svdo** [comment]
After removing most of the aliases from `deps.edn`, removing `.lsp/config.edn` and `.clj-kondo/config.edn`, and reducing the `:client` alias to this, it still happens:

```clojure
            :client { :extra-deps { no.cjohansen/portfolio     {:mvn/version "2023.12.07"}}}
```

When I then also remove the portfolio dependency, the file is not created anymore.

**Comment by borkdude** [comment]
Does it happen with an empty deps edn with only that dependency on the top level? This is what I did above and could not reproduce it that way

**Comment by borkdude** [comment]
Btw, I'm getting this when I try to use that dep now:

```
$ clojure -Spath -Sdeps '{:deps {no.cjohansen/portfolio {:mvn/version "2023.12.07"}}}'
Error building classpath. Could not transfer artifact no.cjohansen:phosphor-clj:jar:2023.07.07 from/to clojars (https://repo.clojars.org/): status code: 416, reason phrase: Range Not Satisfiable (416)
```

**Comment by svdo** [comment]
This is the most minimal reproduction that I could come up with: https://github.com/svdo/repro. Clone, run `clojure-lsp diagnostics`, and the empty file gets created on my machine.

Relevant versions:

```
$ clojure-lsp --version
clojure-lsp 2024.03.01-11.43.56-nightly
clj-kondo 2024.02.13-SNAPSHOT
```

**Comment by borkdude** [comment]
Thanks, that's helpful

**Comment by borkdude** [comment]
I'm still getting:

```
$ clojure -Spath -A:client
Error building classpath. Could not transfer artifact no.cjohansen:phosphor-clj:jar:2023.07.07 from/to clojars (https://repo.clojars.org/): status code: 416, reason phrase: Range Not Satisfiable (416)
```

**Comment by svdo** [comment]
I just did `rm -rf ~/.m2/repository/no` which  removes downloaded phosphor-clj and portfolio. Then I do `clojure -Spath -A:client` and for me it seems to work:

```
Downloading: no/cjohansen/portfolio/2023.12.07/portfolio-2023.12.07.pom from clojars
Downloading: no/cjohansen/phosphor-clj/2023.07.07/phosphor-clj-2023.07.07.pom from clojars
Downloading: no/cjohansen/phosphor-clj/2023.07.07/phosphor-clj-2023.07.07.jar from clojars
Downloading: no/cjohansen/portfolio/2023.12.07/portfolio-2023.12.07.jar from clojars
src:[...]
```

Could it be that one of the servers is misbehaving? IIRC I sometimes see that error pop up in CI for various dependencies, and retrying it tends to solve it.

**Comment by borkdude** [comment]
Yes, it seems to work after I do that too.

```
borkdude@m1 /tmp/repro (main) $ clj-kondo --lint $classpath --dependencies --copy-configs
Configs copied:
- .clj-kondo/no.cjohansen/portfolio
borkdude@m1 /tmp/repro (main?) $ ls .clj-kondo/no.cjohansen/portfolio
config.edn
borkdude@m1 /tmp/repro (main?) $ cat .clj-kondo/no.cjohansen/portfolio/config.edn
borkdude@m1 /tmp/repro (main?) $
```

The logical explanation for this would be that there is an empty `config.edn` in the `clj-kondo.exports` section and this is true:

$ zipinfo /Users/borkdude/.m2/repository/no/cjohansen/portfolio/2023.12.07/portfolio-2023.12.07.jar
```
-rw----     2.0 fat        0 bl defN 23-Dec-07 09:55 clj-kondo.exports/no.cjohansen/portfolio/
-rw----     2.0 fat        0 bl defN 23-Dec-07 09:55 clj-kondo.exports/no.cjohansen/
-rw----     2.0 fat        0 bl defN 23-Dec-07 09:55 clj-kondo.exports/
-rw----     2.0 fat        0 bl defN 23-Oct-13 20:53 clj-kondo.exports/no.cjohansen/portfolio/config.edn
``` 

The solution would be to not do this. cc @cjohansen

**Comment by cjohansen** [comment]
Wow, that is certainly not there intentionally. It's not in the repo. I'll cut a new release without it, thanks for the digging!

**Comment by borkdude** [comment]
Thank you!

**Comment by cjohansen** [comment]
Fixed by a proper clj-kondo config in 2024.03.15 🥳
